src/dmd/mars.d

```diff
@@ -2614,124 +2614,150 @@ private void reconcileCommands(ref Param params, size_t numSrcFiles)
 }
 
 /**
-Creates the list of modules based on the files provided
+Creates the module based on the file provided
 
-Files are dispatched in the various arrays
+The file is dispatched in one of the various arrays
 (global.params.{ddocfiles,dllfiles,jsonfiles,etc...})
-according to their extension.
-Binary files are added to libmodules.
+according to its extension.
+If it is a binary file, it is added to libmodules.
 
 Params:
-  files = File names to dispatch
+  file = File name to dispatch
   libmodules = Array to which binaries (shared/static libs and object files)
                will be appended
 
 Returns:
-  An array of path to D modules
+  A D module
 */
-Modules createModules(ref Strings files, ref Strings libmodules)
+Module createModule(const(char)* file, ref Strings libmodules)
 {
-    Modules modules;
-    modules.reserve(files.dim);
-    bool firstmodule = true;
-    for (size_t i = 0; i < files.dim; i++)
+    const(char)[] name;
+    version (Windows)
     {
-        const(char)[] name;
-        version (Windows)
+        file = toWinPath(file);
+    }
+    const(char)[] p = file.toDString();
+    p = FileName.name(p); // strip path
+    const(char)[] ext = FileName.ext(p);
+    if (ext)
```

Comment:
please invert the order of the order of these and early return of the (current) else branch.

Reply:
I am afraid I am not entirely sure what you're referring to. This is the exact workflow that `mars.d` had, I just extracted this from `createModules` and created a separate function.

Reply:
As in:
```
if (!ext)
{
     if (!p.length)
     {
           error(Loc.initial, "invalid file name '%s'", file);
           fatal();
     }
     auto id = Identifier.idPool(p);
     return new Module(file.toDString, id, global.params.doDocComments, global.params.doHdrGeneration);
}
...
```